server/portal/apps/auth/models.py

```diff
@@ -71,11 +71,19 @@ def client(self):
         :return: Tapis client using refresh token.
         :rtype: :class:Tapis
         """
-        return Tapis(base_url=getattr(settings, 'TAPIS_TENANT_BASEURL'),
+        client = Tapis(base_url=getattr(settings, 'TAPIS_TENANT_BASEURL'),
                      client_id=getattr(settings, 'TAPIS_CLIENT_ID'),
                      client_key=getattr(settings, 'TAPIS_CLIENT_KEY'),
                      access_token=self.access_token,
                      refresh_token=self.refresh_token)
+        
+        if self.expired:
+            client.refresh_tokens()
+            self.update(created=int(time.time()),
+                    access_token=client.access_token.access_token,
+                    expires_in=client.access_token.expires_in().total_seconds())
+
+        return client
```

Comment:
Question - Should there be some potential error handling here like we had in the middleware?

Reply:
Yes, I was thinking we can move that error handling here. Thoughts?

Reply:
Yeah that's what I was leaning towards as well since the middleware doesn't really do anything else. That can be removed and we can do some error handling here